File: RefMan/Models/Referencing/ReferenceResult.cs
namespace RefMan.Models.Referencing
{
    public class ReferenceResult : ReferenceBase
    {
        public ReferenceResult(Reference reference)
        {
            Id = reference.Id;
            AccessDate = reference.AccessDate;
            PublishYear = reference.PublishYear;
            Url = reference.Url;
            IconUrl = reference.IconUrl;
            WebpageTitle = reference.WebpageTitle;
            WebsiteName = reference.WebsiteName;
        }

        public string IdString => Id.ToString();
    }
}

File: RefMan/Client/src/models/referencing/ReferenceResult.ts
import { Optional } from "@/types/Optional";

export interface ReferenceResult {
  id: number;

  idString: string;

  accessDate: string;

  publishYear: Optional<Number>;

  websiteName: string;

  webpageTitle: string;

  url: string;

  iconUrl: string;
}